php/class-wp-https-detection.php

```diff
@@ -0,0 +1,140 @@
+<?php
+/**
+ * WP_HTTPS_Detection class.
+ *
+ * @package PWA
+ */
+
+/**
+ * WP_HTTPS_Detection class.
+ */
+class WP_HTTPS_Detection {
+
+	/**
+	 * The cron hook to check HTTPS support.
+	 *
+	 * @var string
+	 */
+	const CRON_HOOK = 'check_https_support';
+
+	/**
+	 * The option name for whether HTTPS is supported.
+	 *
+	 * @var string
+	 */
+	const HTTPS_SUPPORT_OPTION_NAME = 'is_https_supported';
+
+	/**
+	 * The query var key for HTTPS detection, used to verify that the request is from the same origin.
+	 *
+	 * @var string
+	 */
+	const REQUEST_QUERY_VAR = 'https_detection_token';
+
+	/**
+	 * The number passed to the query var.
+	 *
+	 * @var int
+	 */
+	public $query_var_number;
```

Comment:
This is a property, instead of a simple variable, so that the unit test can access it.